### PR TITLE
Project access levels

### PR DESCRIPTION
**Pull request by xdissent**
This PR adds project access level checks to web socket requests and UI controls, and enables "run" level access for collaborators. Read-only collaborators are no longer allowed to run jobs.

### CODE REVIEW

File: client/account/controllers/account.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ function AccountController($scope, $sce) {
       $scope.success('Account removed');
       return;
     }
-    $.ajax(`/api/account/${account.provider}/${account.id}`, {
+    $.ajax('/api/account/' + account.provider + '/' + account.id, {
```
Comment (knownasilya) [inline client/account/controllers/account.js]: How come you changed this?

Comment (xdissent) [inline client/account/controllers/account.js]: To fix `npm test`. Lint was breaking and actual tests never ran. Looks like the travis setup is broken, but they run locally for me now.

**Comment by knownasilya** [comment]
Would love to have some tests for these changes. Think you can do that? If not that's fine, just let me know.

**Comment by knownasilya** [comment]
If you can do tests, just submit another PR. I'm also going to add you as a collaborator :smile:

**Comment by xdissent** [comment]
Absolutely, I'll add some this afternoon. Thanks!

**Comment by xdissent** [comment]
How would you feel about switching to [protractor](http://www.protractortest.org/) for integration tests? It's pretty difficult to test angular stuff with bare `wd`, and protractor still supports testing plain old html of course. Thoughts?

**Comment by knownasilya** [comment]
Totally for it, protractor is like defacto in the angular community right?

**Pull request merged**